src/histolab/filters/image_filters_functional.py

```diff
@@ -0,0 +1,744 @@
+    img : PIL.Image.Image
+        Input image.
+
+    Returns
+    -------
+    np.ndarray
+     Boolean NumPy array where True represents a pixel above Otsu threshold.
+    """
+    img_arr = np.array(img)
+    otsu_thresh = sk_filters.threshold_otsu(img_arr)
+    # TODO UserWarning: threshold_otsu is expected to work correctly only for grayscale images
+    return _filter_threshold(img_arr, otsu_thresh)
+
+
+def local_otsu_threshold(img: PIL.Image.Image, disk_size: int = 3) -> np.ndarray:
+    """Compute local Otsu threshold for each pixel and boolean mask
+    based on pixels being less than the local Otsu threshold.
+
+    Note that the input image must be 2D.
+
+    Parameters
+    ----------
+    img: PIL.Image.Image
+        Input 2-dimensional image
+    disk_size :int, optional (default is 3)
+        Radius of the disk structuring element used to compute
+        the Otsu threshold for each pixel.
+
+    Returns
+    -------
+    np.ndarray
+        NumPy boolean array representing the mask based on local Otsu threshold
+    """
+    if img.size != 2:
+        raise ValueError("Input must be 2D.")
+    img_arr = np.array(img)
+    local_otsu = sk_filters.rank.otsu(img_arr, sk_morphology.disk(disk_size))
+    return local_otsu
+
+
+def filter_entropy(
+    img: PIL.Image.Image, neighborhood: int = 9, threshold: int = 5
+) -> np.ndarray:
+    """Filter image based on entropy (complexity).
+
+    Note that input must be 2D.
+
+    Parameters:
+    ----------
+    img: PIL.Image.Image
+        input 2-dimensional image
+    neighborhood : int, optionale (default is 9)
+        Neighborhood size (defines height and width of 2D array of 1's).
+    threshold: Threshold value.
+
+    Returns
+    -------
+    np.ndarray
+        NumPy boolean array where True represent a measure of complexity.
+    """
+    if img.size != 2:
+        raise ValueError("Input must be 2D.")
+    img_arr = np.array(img)
+    entropy = sk_filters.rank.entropy(img_arr, np.ones((neighborhood, neighborhood)))
+    return _filter_threshold(entropy, threshold)
+
+
+# input of canny filter is a greyscale
+def canny_edges(
+    img: PIL.Image.Image,
+    sigma: float = 1.0,
+    low_threshold: float = 0.0,
+    high_threshold: float = 25.0,
+) -> np.ndarray:
+    """Filter image based on Canny algorithm edges.
+
+    Note that input image must be 2D
+
+    Parameters
+    ----------
+    img : PIL.Image.Image
+        Input 2-dimensional image
+    sigma : float, optional (default is 1.0)
+        Width (std dev) of Gaussian.
+    low_threshold : float, optional (default is 0.0)
+        Low hysteresis threshold value.
+    high_threshold : float, optional (default is 25.0)
+        High hysteresis threshold value.
+
+    Returns
+    -------
+    np.ndarray
+        Boolean NumPy array representing Canny edge map.
+    """
+    if img.size != 2:
+        raise ValueError("Input must be 2D.")
+    img_arr = np.array(img)
+    canny_edges = sk_feature.canny(
```

Comment:
is it mandatory to use kwargs instead of positional args? i would prefere positional here to keep the function short in terms of line and consistency with the orhters